Add language version control commands to Additions page (#1314)
neo4j/docs-cypher: modules/ROOT/pages/deprecations-additions-removals-compatibility.adoc

label:functionality[]
label:new[]

[source, cypher, role="noheader"]
----

> CREATE [COMPOSITE] DATABASE actors DEFAULT LANGUAGE CYPHER 25

Reviewer: ```suggestion
CREATE [COMPOSITE] DATABASE actors SET DEFAULT LANGUAGE CYPHER 25
```
I know the SET is optional but it looks better to me XD

(similar to [my comment](https://github.com/neo4j/docs-cypher/pull/1123#discussion_r2168517271) on the other PR around the SET)

Reviewer: (just beware that the SET isn't available in Cypher 5 when porting things)

----

a| Set the default Cypher version for a standard or composite database when creating it.
The available versions are `CYPHER 25` and `CYPHER 5`.
If not specified, the default language for the database is set to the default language of the DBMS.
For more information, see the following sections in the Operations Manual:

* link:{neo4j-docs-base-uri}/operations-manual/current/database-administration/standard-databases/create-databases/#_create_a_database_with_default_language[Create a database with a `DEFAULT LANGUAGE`].
* link:{neo4j-docs-base-uri}/operations-manual/current/database-administration/composite-databases/create-composite-databases/#composite-databases-default-language[Set the default Cypher version for a composite database].

a|
label:functionality[]
label:new[]

[source, cypher, role="noheader"]
----

> CREATE ALIAS `remote-with-default-language` FOR DATABASE `northwind-graph-2020`
> AT "neo4j+s://location:7687"
> USER alice
> PASSWORD 'example_secret'
> DEFAULT LANGUAGE CYPHER 25

Reviewer: ```suggestion
CREATE ALIAS `remote-with-default-language`
FOR DATABASE `northwind-graph-2020`
  AT "neo4j+s://location:7687"
  USER alice
  PASSWORD 'example_secret'
  DEFAULT LANGUAGE CYPHER 25
```
This might be a nicer formatting 🤔 makes it more visible that the AT, USER, PASSWORD and DEFAULT LANGUAGE clauses are part of the FOR DATABASE 🤔

----

a| Set the default Cypher version for a remote database alias when creating it.
The available versions are `CYPHER 5` and `CYPHER 25`.

> Local database aliases cannot be assigned a default Cypher version; they always have the Cypher version of their target database.

Reviewer: do we want to mention something around constituent aliases? since those cannot have a default language either (but get theirs from their composite database)

For more information, see the link:{neo4j-docs-base-uri}/operations-manual/current/database-administration/aliases/manage-aliases-standard-databases/#_set_a_default_cypher_version_for_remote_database_aliases[Operations Manual -> Set a default Cypher version for remote database aliases].

a|
label:functionality[]
label:new[]
[source, cypher, role="noheader"]
----
ALTER DATABASE movies SET DEFAULT LANGUAGE CYPHER 25
----

a| Alter the default Cypher version of an existing standard or composite database.
The available versions are `CYPHER 25` and `CYPHER 5`.
For more information, see the link:{neo4j-docs-base-uri}/operations-manual/current/database-administration/standard-databases/alter-databases/#alter-database-default-language[Operations Manual -> Alter database default language].

a|
label:functionality[]
label:new[]

[source, cypher, role="noheader"]
----
ALTER ALIAS `remote-with-default-language` SET DATABASE DEFAULT LANGUAGE CYPHER 25
----

a| Alter the default Cypher version of a remote database alias.
The available versions are `CYPHER 25` and `CYPHER 5`.

> It is not possible to alter the default language of a local database alias; they always have the default language of their target database.

Reviewer: same as above, should we mention constituent aliases?

For more information, see the link:{neo4j-docs-base-uri}/operations-manual/current/database-administration/aliases/manage-aliases-standard-databases/#_alter_the_default_cypher_version_of_a_remote_database_alias[Operations Manual -> Alter the default Cypher version of a remote database alias].

a|
label:functionality[]
label:new[]

[source, cypher, role="noheader"]
----
SHOW DATABASES YIELD name, defaultLanguage
----

a| The new return column `defaultLanguage` for the `SHOW DATABASE` command returns the default language of a database.

> For more information, see the link:{neo4j-docs-base-uri}/operations-manual/current/database-administration/standard-databases/listing-databases/#_show_the_default_cypher_version_of_a_database[Operations Manual -> Show the default Cypher version of a database].

Reviewer: do we want to mention that it isn't returned by default but needs to be yielded? or is it enough like this?

a|
label:functionality[]
label:new[]

[source, cypher, role="noheader"]
----
SHOW ALIAS `remote-with-default-language` FOR DATABASE YIELD name, defaultLanguage
----

a| The new return column `defaultLanguage` for the `SHOW ALIAS` command returns the default language of a database alias.

> For more information, see the link:{neo4j-docs-base-uri}/operations-manual/current/database-administration/standard-databases/listing-databases/#_show_the_default_cypher_version_of_a_database[Operations Manual -> Show the default Cypher version of a database].

Reviewer: same here, do we need to mention it's not a default column?

> For more information, see the link:{neo4j-docs-base-uri}/operations-manual/current/database-administration/standard-databases/listing-databases/#_show_the_default_cypher_version_of_a_database[Operations Manual -> Show the default Cypher version of a database].

Reviewer: shouldn't this link to show aliases instead of show databases?

[source, cypher, role="noheader"]
----
CREATE ALIAS `remote-with-default-language`
FOR DATABASE `northwind-graph-2020`
	AT "neo4j+s://location:7687"
	USER alice
	PASSWORD 'example_secret'
	DEFAULT LANGUAGE CYPHER 25
----

a| Set the default Cypher version for a remote database alias when creating it.
The available versions are `CYPHER 5` and `CYPHER 25`.

> Local database aliases and database aliases of composite databases cannot be assigned a default Cypher version; they always have the Cypher version of their target database.

Reviewer: ```suggestion
Local database aliases and database aliases in composite databases cannot be assigned a default Cypher version. Local database aliases always have the Cypher version of their target database and database aliases in composite databases always have the Cypher version of the composite database they belong to.
```

a| Set the default Cypher version for a remote database alias when creating it.
The available versions are `CYPHER 5` and `CYPHER 25`.
Local database aliases and database aliases in composite databases cannot be assigned a default Cypher version.

> Local database aliases always have the Cypher version of their target database and database aliases in composite databases always have the Cypher version of the composite database they belong to.For more information, see the link:{neo4j-docs-base-uri}/operations-manual/current/database-administration/aliases/manage-aliases-standard-databases/#_set_a_default_cypher_version_for_remote_database_aliases[Operations Manual -> Set a default Cypher version for remote database aliases].

Reviewer: ```suggestion
Local database aliases always have the Cypher version of their target database and database aliases in composite databases always have the Cypher version of the composite database they belong to.
For more information, see the link:{neo4j-docs-base-uri}/operations-manual/current/database-administration/aliases/manage-aliases-standard-databases/#_set_a_default_cypher_version_for_remote_database_aliases[Operations Manual -> Set a default Cypher version for remote database aliases].
```

a| Alter the default Cypher version of a remote database alias.
The available versions are `CYPHER 25` and `CYPHER 5`.

> It is not possible to alter the default language of a local database alias or an alias of a composite database; they always have the default language of their target database.

Reviewer: ```suggestion
It is not possible to alter the default language of a local database alias or an alias belonging to a composite database. Local database aliases always have the default language of their target database and aliases belonging to composite databases always have the default language of the composite database.
```

Reviewer: Also, why does the ALTER case say `default language` and the CREATE case say `Cypher version`? feels like they should be aligned? (and aligned with the database ones)

a| Alter the default Cypher version of a remote database alias.
The available versions are `CYPHER 25` and `CYPHER 5`.
It is not possible to alter the default Cypher version of a local database alias or an alias belonging to a composite database.

> Local database aliases always have the Cypher version of their target database and aliases belonging to composite databases always have the default language of the composite database.

Reviewer: ```suggestion
Local database aliases always have the Cypher version of their target database and aliases belonging to composite databases always have the Cypher version of the composite database.
```
